macros.cfg

```diff
@@ -72,4 +73,56 @@ gcode:
   G1 Z0.3 F3000 ; get ready to prime
   G92 E0 ; reset extrusion distance
   G1 Y80 E16 F1200 ; prime nozzle
-  G1 Y100 F15000 ; quick wipe
\ No newline at end of file
+  G1 Y100 F15000 ; quick wipe
+
+[gcode_macro CHANGE_FILAMENT]
+default_parameter_TEMP: 220
+gcode:
+    SAVE_GCODE_STATE NAME=change_filament_state
+    PAUSE ; Pauze the print
+    LOW_TEMP_CHECK T={TEMP} ; Heat up hotend to provided temp or 220 as default as that should work OK with most filaments.
+    M83 ; use relative extrusion distances
+    G92 E0 ; reset extrusion distance
+    G1 E-1 F2700 ; retract filament to prevent oozing
+    PARK ; Send nozzle to park position
+    UNLOAD_FILAMENT ; Call unload macro
+    RESTORE_GCODE_STATE NAME=change_filament_state
+
+# Check if we need to change the hotend temperature
+[gcode_macro LOW_TEMP_CHECK]
+default_parameter_T: 230
+gcode: 
+    {% if printer.extruder.target != 0 %}
+        {% if printer.extruder.temperature < printer.extruder.target %} 
+            M109 S{printer.extruder.target|float} 
+        {% endif %}
+    {% else %}
+        {% if printer.extruder.target < T %}
+            M109 S{T}
+        {% endif %}
+    {% endif %}
+
+# Park the nozzle in a safe area
+[gcode_macro PARK]
+default_parameter_X: 0
+default_parameter_Y: 0
+default_parameter_Z: 20
+gcode:
+    SAVE_GCODE_STATE NAME=park_state
+    M117 Parking toolhead...
+    {% if printer.gcode_move.position.z|float < (printer.toolhead.axis_maximum.z - 20) %} ; Check if we are below maximum tool height
+        G91 ; use relative coordinates
+        G1 Z{Z} F7200 ; raise z axis by Z mm
+    {% else %}
+        G90 ; use absolute coordinates
+        G1 Z{printer.toolhead.axis_maximum.z} F7200 ; bring z axis to max height
+    {% endif %}
+    G90 ; use absolute coordinates
+    G1 X{X} Y{Y} F7200 ; move to park position
```

Comment:
I'd prefer if these could be `X{printer.toolhead.axis_maximum.x / 2} Y{printer.toolhead.axis_maximum.y}` by default. The we can use this macro in START_PRINT, END_PRINT and PAUSE as well. You can use the default() jinja method to check for params instead, and drop the default_parameter_X and Y.

Reply:
I set the default to the back rear and modified the CHANGE_FILAMENT macro to use 0,0 for easy of access to the nozzle.